elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/AggregationDataStoreHelper.java

```diff
@@ -85,22 +93,60 @@ private void splitFilters() {
         havingFilter = constraints.getHavingExpression();
     }
 
-    /**
-     * Gets dimensions and timeDimensions based on relationships and attributes from {@link EntityProjection}.
-     * @param dimensions Empty set of {@link Dimension} objects.
-     * @param timeDimensions Empty set of {@link TimeDimension} objects.
-     */
-    private void resolveDimensionLists(Set<Dimension> dimensions, Set<TimeDimension> timeDimensions) {
-        Set<String> relationshipNames = getRelationships();
-        Set<String> attributeNames = getAttributes(); // time dimensions are under attribute in entity projection
-        for (Dimension dimension : schema.getDimensions()) {
-            if (dimension instanceof TimeDimension && attributeNames.contains(dimension.getName())) {
-                timeDimensions.add((TimeDimension) dimension);
+    //TODO - Add tests in the next PR.
+    private Set<ProjectedTimeDimension> resolveTimeDimensions() {
+        Set<ProjectedTimeDimension> timeDims = new LinkedHashSet<>();
+        //Only attributes can be time dimensions
+        entityProjection.getAttributes().stream().forEach((attribute -> {
+            TimeDimensionColumn timeDim = schema.getTimeDimension(attribute.getName());
```

Comment:
Not sure about this one.  The lambda is more less the entire thing the function does.  If I move it out, it will effectively just move the entire function.

The complicated bit of this is to hydrate the TimeDimension grain from the EntityProjection arguments.  The logic itself seems pretty tiny (in terms of cyclomatic complexity).  Anyway - open to ideas on how to improve this - but not sure what to do in this case.

Reply:
Ok, but I think we can at least match the filtering logic with `resolveNonTimeDimensions()` method. That is
```
timeDims = getAttributes().stream()
    .map(columnName -> schema.getDimension(columnName))
    .filter(Objects::nonNull)
    .filter(column -> column instanceof TimeDimensionColumn)
...
```